### ISSUE TITLE
Quickstart SE Gradle build improvements

### ISSUE THREAD

**Issue by m0mus** [oracle/helidon#2186]
1. Gradle requires an explicit setting to let it use JUnit5, having the Jupiter dependencies in the class path is not enough as it is in Maven ->

```
test {
    useJUnitPlatform()
}
```

2. The build file does not define a Jupiter engine which will make tests fail. Also, the `testCompileOnly` configuration won’t work either, it must be `testImplementation` instead.
Dependencies defined by creating a project using the helidon-se archetype (2.0.1) differ from those found in the build file

```
dependencies {
    implementation platform("io.helidon:helidon-dependencies:${helidonVersion}")
    implementation 'io.helidon.webserver:helidon-webserver'
    implementation 'io.helidon.media:helidon-media-jsonp'
    implementation 'io.helidon.config:helidon-config-yaml'
    implementation 'io.helidon.health:helidon-health'
    implementation 'io.helidon.health:helidon-health-checks'
    implementation 'io.helidon.metrics:helidon-metrics'

    testImplementation 'org.junit.jupiter:junit-jupiter-api'
    testImplementation 'io.helidon.webclient:helidon-webclient'
    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
}
```

3. There’s no need to compute a Class-Path manifest entry. The `application` plugin can launch the main class with the right class path configured. The `shadow` plugin can be used to generate an uberjar.

Please see https://gist.github.com/aalmiray/9fb34beccbce21e9d4475a38de506c2c for more information. Gradle requires a minimum of 2 files (settings.gradle & build.gradle). I prefer defining static properties in a 3rd file named gradle.properties, though it’s possible to combine all 3 files in a single build.gradle and leave just one instruction (the rootProject’s name) in settings.gradle.

One can invoke from the command line the following:

```
$ gradle run
```

Runs the main class. Akin to invoking `mvn compile exec:java`.

```
$ gradle shadowJar
```

Generates the uberjar. Shadow is aware of the Main-Class manifest entry. Akin to invoking `mvn package` when shade is configured.

```
$ gradle runShadow
```

Runs the shaded application. This is a combination of the 2 previous commands. The shaded/shadowed jar can be executed with

```
$ java -jar build/libs/helidon-sample-1.0.0-SNAPSHOT-all.jar
```

**Comment by aalmiray** [comment]
These changes apply to Helidon SE only. A different set of changes are required for Helidon MP.

**Comment by barchetta** [comment]
Some of this addressed by PR #2214

**Comment by barchetta** [comment]
Since the PR fixed the most serious of the issues listed here I'm going to close this and use issue #2219 to track additional improvements to the quickstart example gradle files.
